pkg/webhook/istiocontrolplane/server.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package istiocontrolplane
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	"istio.io/pkg/log"
+)
+
+func Add(mgr manager.Manager) error {
+	return add(mgr)
+}
+
+func add(mgr manager.Manager) error {
+	log.Info("setting up webhook server")
+	webhookServer := mgr.GetWebhookServer()
+
+	webhookServer.Register("/validate-v1alpha2-istiocontrolplane",
+		&webhook.Admission{Handler: &iscpValidator{}})
```

Comment:
We should be setting all the webhook fileds, e.g.:
* Rules
* FailurePolicy
* Type

Reply:
seems like there is no place to specify Rules/Types during registration. Can you point me to an example, I think API of controller-runtime changes quite a bit

Reply:
this is doc I found from kubebuilder that might be related: https://book.kubebuilder.io/migration/v1vsv2.html